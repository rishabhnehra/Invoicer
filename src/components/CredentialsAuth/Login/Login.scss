.Login {
    position: relative;

    header {
        display: flex;
        justify-content: center;
        font-size: 25px;
        font-weight: bold;
        color: rgba(0,0,0,0.4);
        margin: 8px 0 32px 0;
    }

    hr {
        margin: 32px 0;
    }

    hr::after {
        content: 'or';
        font-size: 15px;
        padding: 0 5px;
        background-color: #E2F3F5;
        position: absolute;
        bottom: 25.5%;
        left: 50%;
        transform: translateX(-50%)
    }

    .form-item {
        padding: 8px 0;
        
        .TextInput {
            box-shadow: none;
        }

        &__submit {
            margin: 32px 0 20px 0;
            height: 49px;

            .button {
                font-weight: bold;
                font-size: 18px;
                height: 100%;
                width: 100%;
                background-color: #fff;
                color: #000;
                box-shadow: none;
            }
        }

        &__google-login {
            margin: 20px 0 32px 0;
            height: 49px;

            .button {
                font-weight: bold;
                font-size: 18px;
                height: 100%;
                width: 100%;
                background-color: #000;
                color: #fff;
                box-shadow: none;
            }
        }
    }
}