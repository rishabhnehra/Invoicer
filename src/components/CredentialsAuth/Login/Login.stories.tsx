import {Story, Meta} from '@storybook/react/types-6-0';

import { Login as LoginComponent } from '../index';

export default {
    title: 'Components/Credentials Auth',
    component: LoginComponent
} as Meta;

const Template: Story = (args) => <LoginComponent {...args} />

export const Login = Template.bind({});
