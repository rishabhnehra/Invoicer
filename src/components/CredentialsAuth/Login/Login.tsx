import { Cards, TextInput, Button } from '../../index';
import './Login.scss';

export type LoginProps = {
    onClick?: () => void;
    onChange?: () => void;
    onClickGoogleHandler?: () => void;
}

export const Login: React.FC<LoginProps> = ({
    onClickGoogleHandler,
    onClick,
    onChange
}) => <Cards.BlankCard className="Login">
        <header>
            <h3>Login</h3>
        </header>
        <form>
            <div className="form-item">
                <TextInput type="email" placeholder="Email" onChange={onChange} />
            </div>
            <div className="form-item">
                <TextInput type="password" placeholder="Password" onChange={onChange} />
            </div>
            <div className="form-item__submit">
                <Button onClick={onClick}>Continue</Button>
            </div>
        </form>
        <hr />
        <div className="form-item__google-login">
            <Button onClick={onClickGoogleHandler}>Login with Google</Button>
        </div>
    </Cards.BlankCard>