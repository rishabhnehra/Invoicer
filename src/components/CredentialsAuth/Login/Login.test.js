import renderer from 'react-test-renderer';
import { Login } from './Login';

describe("<Login />", () => {
    it("snapshot check", () => {
        const tree = renderer.create(<Login />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});