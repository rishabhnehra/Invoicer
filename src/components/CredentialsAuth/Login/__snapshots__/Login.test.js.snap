// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> snapshot check 1`] = `
<div
  className="BlankCard Login"
>
  <header>
    <h3>
      Login
    </h3>
  </header>
  <form>
    <div
      className="form-item"
    >
      <input
        className="TextInput"
        placeholder="Email"
        type="email"
      />
    </div>
    <div
      className="form-item"
    >
      <input
        className="TextInput"
        placeholder="Password"
        type="password"
      />
    </div>
    <div
      className="form-item__submit"
    >
      <button
        className="button primary"
        style={
          Object {
            "backgroundColor": undefined,
          }
        }
      >
        Continue
      </button>
    </div>
  </form>
  <hr />
  <div
    className="form-item__google-login"
  >
    <button
      className="button primary"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    >
      Login with Google
    </button>
  </div>
</div>
`;
